### ISSUE TITLE
Alina: web-apps, 5 weeks

### ISSUE THREAD

**Issue by alinamarasca** [HackYourFutureBelgium/class-13-14#343]
> [module repository](https://github.com/alinamarasca/web-apps)

## Study Plan
* figure out postman
* try some projects which use API
Week 1

- [ x] started the exercises

#### I Need Help With:

..with understanding if I understand.

I followed few tutorials about Postman and usage of API's for simple apps (like showing random dogs/cats pictures) and it seems to be very simple. At the same time, I don't feel myself confident in the topic.
I guess, only practice can help me.

I am planing to re-watch Monday session with Yoshi and make the project he shows.

#### What went well?
???
#### What went less well?
???
#### Lessons Learned
???

**Comment by colevandersWands** [comment]
> At the same time, I don't feel myself confident in the topic.

we dumped everything on you in the first week so you'd have more time to study the whole picture.  no worries if it hasn't clicked yet.

**Comment by colevandersWands** [comment]
> writing basic authentication, at this pint

this is a tricky thing, no need to master it right away. it's this early in the course so you have time to understand it by the final project

> i guess, modules in this context are confusing me - I am lost again what should be where

is it how module.exports and require work? or about how to split your code between the different folders?

**Comment by alinamarasca** [comment]
@colevandersWands just noticed the comment.  
> basic auth 
Do i even need to master it? It looks like it is not  even used nowadays.  It is mostly token or JWT and OAuth(whatever it means). Does front-end er even do those things?

>modules
hi think, i should have said  components. how do i split authorization into components?

**Comment by colevandersWands** [comment]
> Do i even need to master it? It looks like it is not even used nowadays. It is mostly token or JWT and OAuth(whatever it means). Does front-end er even do those things?

it's good to know about, but don't need to master it.  even if other things are newer there's still a lot of old stuff out there and you never know what they'll use at your first job

**Comment by colevandersWands** [comment]
> hi think, i should have said components. how do i split authorization into components?

you mean in the fronend?

**Comment by colevandersWands** [comment]
@yoshimalaise , any tips on connecting auth to frontend?

**Comment by yoshimalaise** [comment]
Hi @alinamarasca,

So the short answer is: you don't.

The longer answer: Your components should only be concerned about what the application looks like and how the end user can interact with your system. So the authentication isn't really something that your components need to worry about, instead the authorization/authentication aspects of your code should be moved to your business logic.

You would need to provide a login method that will call your backend to request a session token and upon receiving this token it should be stored somewhere in the state of your application. From that point on your components can keep calling function that will perform fetch requests just like you have been doing during the previous modules, the only small piece that is different is that from now on your api call functions will read your session from the state and include it in all future fetch requests. This way all future requests will be authenticated, but your components do not need to worry about it at all.

**Comment by yoshimalaise** [comment]
This is definitely something we will recap during the session this week so feel free to prepare any questions about this that you might have.

**Comment by alinamarasca** [comment]
my components won't care, but I do. and it really bothers me I can't figure out the whole picture..i kind of need to organize the storage in my head.

**Comment by yoshimalaise** [comment]
Did the extra explanation during last week's session clear things up for you?

**Comment by alinamarasca** [comment]
no, not really. I think, I will have to find my own way to explain it to myself. But that's normal :wink:
